src/ark/segmentation/ez_seg/composites.py

```diff
@@ -73,28 +76,34 @@ def composite_builder(
                 fov_data, composite_array, images_to_subtract, image_type, composite_method
             )
 
-        # Create the fov dir within the composite dir
-        composite_fov_dir = composite_directory / fov
-        composite_fov_dir.mkdir(parents=True, exist_ok=True)
+        if composite_directory:
+            # Create the fov dir within the composite dir
+            composite_fov_dir = composite_directory / fov
+            composite_fov_dir.mkdir(parents=True, exist_ok=True)
 
-        # Save the composite image
-        image_utils.save_image(
-            fname=composite_directory / fov / f"{composite_name}.tiff",
-            data=composite_array.astype(np.float32)
-        )
+            # Save the composite image
+            image_utils.save_image(
+                fname=composite_directory / fov / f"{composite_name}.tiff",
+                data=composite_array.astype(np.uint32)
+            )
 
+        composite_images[fov] = composite_array.astype(np.float32)
+        
     # Write a log saving composite builder info
-    variables_to_log = {
-        "image_data_dir": image_data_dir,
-        "fov_list": fov_list,
-        "images_to_add": images_to_add,
-        "images_to_subtract": images_to_subtract,
-        "image_type": image_type,
-        "composite_method": composite_method,
-        "composite_directory": composite_directory,
-        "composite_name": composite_name,
-    }
-    log_creator(variables_to_log, log_dir, f"{composite_name}_composite_log.txt")
+    if log_dir:
+        variables_to_log = {
+            "image_data_dir": image_data_dir,
+            "fov_list": fov_list,
+            "images_to_add": images_to_add,
+            "images_to_subtract": images_to_subtract,
+            "image_type": image_type,
+            "composite_method": composite_method,
+            "composite_directory": composite_directory,
+            "composite_name": composite_name,
+        }
+        log_creator(variables_to_log, log_dir, f"{composite_name}_composite_log.txt")
```

Comment:
The ezseg functions only save the composites, they don't return them so I would have to change the ezseg notebook to handle a return (not a big deal, but unnecessary when it's not even used). Since the only case we would need the composites returned is for the internal function call inside generate_signal_masks(), I didn't think it was needed.